### ISSUE TITLE
p-tuning or prefix tuning or prompt tuning code given does not work during inference

### ISSUE THREAD

**Issue by chandar-l** [huggingface/peft#2131]
### System Info

In https://github.com/huggingface/peft/blob/main/docs/source/task_guides/prompt_based_methods.md, the p-tuning method (also referred to in the hugging faces documentation page: https://huggingface.co/docs/peft/task_guides/prompt_based_methods) or the prefix tuning method or the prompt tuning method fails at the last step when we try to do a model.generate.  We get an error message saying that : 

The expanded size of the tensor (37) must match the existing size (57) at non-singleton dimension 2.  Target sizes: [16, 37, 37].  Tensor sizes: [16, 1, 57]

### Who can help?

@jpodivin 

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder
- [ ] My own task or dataset (give details below)

### Reproduction

Just following instructions given in https://github.com/huggingface/peft/blob/main/docs/source/task_guides/prompt_based_methods.md for p-tuning method or the prefix tuning method or the prompt tuning method

### Expected behavior

Expected the model to generate complaint as per the documentation when running the last cell of the notebook:

with torch.no_grad():
    inputs = {k: v.to(device) for k, v in inputs.items()}
    outputs = model.generate(input_ids=inputs["input_ids"], max_new_tokens=10)
    print(tokenizer.batch_decode(outputs.detach().cpu().numpy(), skip_special_tokens=True))

 But it fails with error mentioned above.

**Comment by BenjaminBossan** [comment]
Could you please tell me what versions of transformers, tokenizers, and PEFT you're using?

**Comment by chandar-l** [comment]
Hi Benjamin,I am using the following versions:
peft                               0.13.0tokenizers                         0.20.0transformers                       4.45.1
An update after posting my issue in GitHub.
If I change the inference code from
 outputs = model.generate(input_ids=inputs[\"input_ids\"], max_new_tokens=10)

to
 outputs = model.generate(input_ids=inputs[\"input_ids\"], max_new_tokens=10, use_cache=False)

it starts working fine.  Is this expected?
Regards,Chandar

    On Monday, October 7, 2024 at 06:41:18 PM GMT+5:30, Benjamin Bossan ***@***.***> wrote:  
 
 


Could you please tell me what versions of transformers, tokenizers, and PEFT you're using?

**Comment by BenjaminBossan** [comment]
Thanks for the extra info. I believe this is already fixed. Could you try installing PEFT from source and check if the error still occurs?

**Comment by chandar-l** [comment]
Thanks - it worked with latest 0.13.1 version from the github
    On Monday, October 7, 2024 at 07:35:10 PM GMT+5:30, Benjamin Bossan ***@***.***> wrote:  
 
 


Thanks for the extra info. I believe this is already fixed. Could you try installing PEFT from source and check if the error still occurs?

**Comment by BenjaminBossan** [comment]
Great to hear. I'll close the issue then, feel free to re-open if something new comes up.
